app/vminsert/influx/request_handler.go

```diff
@@ -105,9 +106,15 @@ func insertRows(at *auth.Token, db string, rows []parser.Row, mayOverrideAccount
 			ic.Labels = ic.Labels[:len(ic.Labels)-1]
 			ic.AddLabel("", metricGroup)
 			ic.MetricNameBuf = storage.MarshalMetricLabelRaw(ic.MetricNameBuf[:metricNameBufLen], placeholderLabel)
-			storageNodeIdx := ic.GetStorageNodeIdx(&atCopy, ic.Labels)
-			if err := ic.WriteDataPointExt(&atCopy, storageNodeIdx, ic.MetricNameBuf, r.Timestamp, f.Value); err != nil {
-				return err
+			storageNodeGroupIDs := ic.GetStorageNodeGroupIds(&atCopy, ic.Labels)
+			for _, storageNodeGroupID := range storageNodeGroupIDs {
+				if err := ic.WriteDataPointExt(at, storageNodeGroupID, ic.MetricNameBuf, r.Timestamp, f.Value); err != nil {
+					if ic.AllReplicationGroupsFailed(storageNodeGroupID.Group) {
+						logger.Errorf("All replication groups have failed")
```

Comment:
It is better starting log messages with small letter, since logging subsystem may prepend the message with another text like the following: `some prefix here: actual log message`.